src/main/java/org/opensearch/timeseries/TimeSeriesAnalyticsPlugin.java

```diff
@@ -1672,6 +1679,17 @@ public List<NamedXContentRegistry.Entry> getNamedXContent() {
             );
     }
 
+    @Override
+    public Collection<SystemIndexDescriptor> getSystemIndexDescriptors(Settings settings) {
+        List<SystemIndexDescriptor> systemIndexDescriptors = new ArrayList<>();
+        systemIndexDescriptors.add(new SystemIndexDescriptor(ALL_AD_RESULTS_INDEX_PATTERN, "Time Series Analytics Results index pattern"));
+        systemIndexDescriptors
+            .add(new SystemIndexDescriptor(ALL_AD_DETECTOR_INDEX_PATTERN, "Time Series Analytics Detector index pattern"));
+        systemIndexDescriptors.add(new SystemIndexDescriptor(CHECKPOINT_INDEX_NAME, "Time Series Analytics Checkpoints index"));
+        systemIndexDescriptors.add(new SystemIndexDescriptor(DETECTION_STATE_INDEX, "Time Series Analytics Detection State index"));
+        return systemIndexDescriptors;
```

Comment:
Can you add the following indices?

ForecastCommonName.FORECAST_CHECKPOINT_INDEX_NAME: Forecast Checkpoints index
ForecastCommonName.FORECAST_STATE_INDEX: Forecast state index
CommonName.JOB_INDEX: Time Series Analytics job index

I am leaving out ForecastIndexManagement.FORECAST_RESULT_HISTORY_INDEX_PATTERN as default forecast result index is a user index, not a system index (as defined in the security plugin).

Reply:
Added these indices.